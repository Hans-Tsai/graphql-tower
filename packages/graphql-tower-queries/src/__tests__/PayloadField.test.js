import faker from 'faker';
import { graphql, GraphQLInt, GraphQLSchema, GraphQLObjectType } from 'graphql';
import { Query, PayloadField } from '../';

describe('PayloadField', () => {
  it('snapshot', async () => {
    const payloadField = new PayloadField();
    expect(payloadField).toMatchSnapshot();
  });

  it('successfully query', async () => {
    const resolve = jest.fn((payload, { id }) => id);
    const Node = class extends Query {
      type = GraphQLInt;
      resolve = resolve;
    };

    const fieldId = faker.random.number();
    const tempId = faker.random.number();
    const value = faker.random.number();

    const schema = new GraphQLSchema({
      query: new GraphQLObjectType({
        name: 'Query',
        fields: {
          node: {
            type: new GraphQLObjectType({
              name: 'Node',
              fields: {
                field: new Node({ id: new PayloadField() }),
                custom: new Node({ id: new PayloadField('tempId') }),
                handler: new Node({ id: new PayloadField(({ tempId: id }) => id) }),
                value: new Node({ id: value }),
              },
            }),
            resolve: () => ({ id: fieldId, tempId }),
          },
        },
      }),
    });

    const result = await graphql(schema, 'query { node { field custom handler value } }');
    expect(result.data.node).toEqual({
      field: fieldId, custom: tempId, handler: tempId, value,
    });
  });
});
