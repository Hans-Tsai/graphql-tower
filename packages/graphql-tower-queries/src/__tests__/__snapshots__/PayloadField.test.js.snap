// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayloadField snapshot 1`] = `
PayloadField {
  "fieldName": undefined,
}
`;
